library/cwm/src/lib/cwm/widget.rb

```diff
@@ -630,4 +643,182 @@ class RichText < AbstractWidget
 
     include ValueBasedWidget
   end
+
+  # Tab widget, usefull only with {CWM::Tabs}
+  # @see tabs example for usage
+  class Tab < CustomWidget
+    attr_accessor :initial
+
+    # content of tab, can contain another widgets
+    abstract_method :contents
+
+    def cwm_definition
+      super.merge(
+        "widgets"       => cwm_widgets,
+        "custom_widget" => Yast::CWM.PrepareDialog(cwm_contents, cwm_widgets)
+      )
+    end
+
+    # get cwm style of widget definitions
+    # @note internal api only used as gate to communicate with CWM
+    def cwm_widgets
+      return @cwm_widgets if @cwm_widgets
+
+      widgets = nested_widgets
+      names = widgets.map(&:widget_id)
+      definition = Hash[widgets.map { |w| [w.widget_id, w.cwm_definition] }]
+      @cwm_widgets = Yast::CWM.CreateWidgets(names, definition)
+    end
+
+  protected
+
+    # defines name of tab header
+    # It is recommended to rewrite it
+    def label
+      textdomain "base"
+
+      _("Tab")
+    end
+
+    # help that is result of used widget helps.
+    # If overwritting, do not forget to use `super`, otherwise widget helps will
+    # be missing
+    def help
+      Yast::CWM.MergeHelps(nested_widgets.map(&:cwm_definition))
+    end
+  end
+
+  # useful to have tabs as widget. It contained {CWM::Tab} with its content
+  # @see examples/object_api_tabs.rb
+  class Tabs < CustomWidget
+    # param [Array<CWM::Tab>] tabs to be shown
+    def initialize(*tabs)
+      @tabs = tabs
+      @current_tab = nil
+      self.handle_all_events = true
+    end
+
+    # gets visual order of tabs
```

Comment:
it is a bit mistery, but when I move it to private it stop working without anything in logs